src/app/App.tsx

```diff
@@ -86,6 +86,50 @@ export function App(): React.JSX.Element {
     });
   }
   
+  async function searchSnippetList(snippetName: string) {
+    try {
+      const searchedAssets = await new Pieces.SearchApi().fullTextSearch({ query: snippetName });
+      
+      // Check if there are no matching snippets 
+      if (searchedAssets.iterable.length === 0) {
+        return 'No matching snippets found';
+      }
+
+      // get the "ID" or identifier of the first match on the string you passed in as the query:
+      let firstSearchMatchAssetIdentifier = searchedAssets.iterable[0].identifier;
+  
+      let matchName: String;
+  
+      // take that identifier to get your assets name using the Pieces.AssetApi()
+      const asset = await new Pieces.AssetApi().assetSnapshot({asset: firstSearchMatchAssetIdentifier});
+  
+      // assign that name to the matchName variable:
+      matchName = asset.name;
+      console.log("the matchName is" + matchName);
+  
+      // then you can do whatever you would like with that match:   
+      return matchName;
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const [searchTerm, setSearchTerm] = useState('')
+  const [searchResult, setSearchResult] = useState('')
+  const handleSearch = async (event) => {
+    event.preventDefault()
+    const searchTerm = event.target.elements['search-term'].value
+    if (!searchTerm) {
+      setSearchResult('');
+      return;
+    }
+    setSearchTerm(searchTerm );
+    let result = await searchSnippetList(searchTerm);
+    console.log("The Result is "+ result);
+    console.log(searchTerm);
```

Comment:
ill handle this here, but a good note in the future is to try and minimize console.logs when finalizing a PR, removing some unnecessary ones that were used in development. 👍

Reply:
Sorry. I forgot to remove those! 

I'll keep this in mind.